File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;
/**
 * 品牌接口
 *
 * @author dxf
 */

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

public interface BrandService {

    //查询所有
    public List<TbBrand> findAll();

    //分页查询
    public PageResult findPage(Integer currentPage, Integer pageSize);

    //保存brand
    public void saveBround(TbBrand brand);


    //修改brand
    public void changeBrand(TbBrand brand);

    //删除brand
    public void deleteBrand(Long[] ids);

    //模糊查询
    public PageResult findPage(Integer currentPage, Integer pageSize, TbBrand brand);

    //查询所有的品牌信息封装到map集合中并整个变为list集合
    public List<Map> selectOptionList();
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js

//服务层
app.service("brandService", function ($http) {
    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };

    //分页
    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size + '');
    }


    //保存brand
    this.saveBrand=function (entity) {
        return $http.post('../brand/saveBrand.do?', entity);
    }

    //修改数据
    this.updateBrand=function (entity) {
        return $http.post('../brand/updateBrand.do', entity)
    }

    //删除数据
    this.deleteBrand=function (ids) {
        return $http.post('../brand/deleteBrand.do?ids=' + ids + '')
    }

    //条件查询
    this.searchBrand=function (page,size,searchEntity) {
        return $http.post('../brand/searchPage.do?page=' + page + '&size=' + size + '', searchEntity)
    }






});